lib/bluetooth_device.js

```diff
@@ -14,9 +17,27 @@
   BluetoothDevice.prototype.notifyCharacteristic = null;
   BluetoothDevice.prototype._isNotificationsStarted = false;
 
+  /**
+   * Send data to BLE device in bytes.
+   * @param {String|Uint8Array} data Data to be sent,
+   * accept Uint8Array or String in hex format.
+   * @throws "Data can't be empty" if data is null|undfined
+   * @throws "Unsupported data type" if data is not String|Uint8Array
+   */
   BluetoothDevice.prototype.send = function(data) {
-    data = this._parseHexString(data);
-    this.writeCharacteristic.writeValue(data);
+    if (data instanceof Uint8Array) {
+      this.writeCharacteristic.writeValue(data.buffer);
+    }
+    else if (data instanceof String) {
+      this.writeCharacteristic.writeValue(this._parseHexString(data));
+    }
```

Comment:
But what for cases that invalid data passed in?
If we don't throw exception, it will just failed silently, make it hard to debug.